File: go_backend/models/espacio.go
package models

type Espacio struct {
	ID          string  `json:"id"`
	UsuarioID   string  `json:"usuario_id"`
	Nombre      string  `json:"nombre"`
	NombreNorm  string  `json:"-"`
	Descripcion string  `json:"descripcion,omitempty"`
	PadreID     *string `json:"padre_id,omitempty"`
	CreatedAt   Time    `json:"created_at"`
	UpdatedAt   Time    `json:"updated_at"`
}

File: flutter_app/lib/core/models/espacio.dart
class Espacio {
  final String id;
  final String nombre;
  final String descripcion;
  final String? padreId;
  final bool tieneHijos;

  Espacio({
    required this.id,
    required this.nombre,
    this.descripcion = '',
    this.padreId,
    this.tieneHijos = false,
  });

  factory Espacio.fromJson(Map<String, dynamic> json) => Espacio(
        id: json['id'] as String,
        nombre: json['nombre'] as String,
        descripcion: json['descripcion'] as String? ?? '',
        padreId: json['padre_id'] as String?,
        tieneHijos: json['tiene_hijos'] as bool? ?? false,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'nombre': nombre,
      };
}
